Subject: patch for arm
From: Philip Blundell <Philip dot Blundell at pobox dot com>
To: libc-hacker at sources dot redhat dot com
Content-Type: text/plain
Content-Transfer-Encoding: 7bit
Date: 22 Jul 2002 20:12:06 +0100
Message-Id: <1027365126.725.10.camel@kc>
Mime-Version: 1.0

The mainline seems to need this patch in order to compile.

p.

2002-07-22  Philip Blundell  <philb@gnu.org>

	* sysdeps/unix/sysv/linux/arm/errlist.c: Remove extra weak alias
	definiton of _old_sys_nerr.  Define _old_sys_errlist as strong
	alias.

Index: sysdeps/unix/sysv/linux/arm/errlist.c
===================================================================
RCS file: /cvs/glibc/libc/sysdeps/unix/sysv/linux/arm/errlist.c,v
retrieving revision 1.4
diff -u -p -r1.4 errlist.c
--- glibc/sysdeps/unix/sysv/linux/arm/errlist.c	6 Jul 2001 04:56:13 -0000	1.4
+++ glibc/sysdeps/unix/sysv/linux/arm/errlist.c	22 Jul 2002 19:10:57 -0000
@@ -1,4 +1,4 @@
-/* Copyright (C) 1998, 2000 Free Software Foundation, Inc.
+/* Copyright (C) 1998, 2000, 2002 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -38,10 +38,9 @@ extern const char *const *__old_sys_errl
 const int __old_sys_nerr = OLD_ERRLIST_SIZE;
 
 strong_alias (__old_sys_nerr, _old_sys_nerr);
-weak_alias (__old_sys_nerr, _old_sys_nerr)
 compat_symbol (libc, __old_sys_nerr, _sys_nerr, GLIBC_2_0);
 compat_symbol (libc, _old_sys_nerr, sys_nerr, GLIBC_2_0);
-weak_alias (__old_sys_errlist, _old_sys_errlist);
+strong_alias (__old_sys_errlist, _old_sys_errlist);
 compat_symbol (libc, __old_sys_errlist, _sys_errlist, GLIBC_2_0);
 compat_symbol (libc, _old_sys_errlist, sys_errlist, GLIBC_2_0);
 #endif




